### PR TITLE
SB-1475: add specific liquibase changeset method to change property type from 'date' to 'datetime'

### PR DESCRIPTION
**Pull request by evsikov**
…rom 'date' to 'datetime' while saving the existing data

### CODE REVIEW

File: src/main/java/org/unbrokendome/liquibase/orientdb/statement/ChangePropertyTypeDateToDatetimeStatement.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package org.unbrokendome.liquibase.orientdb.statement;
+
+import liquibase.statement.AbstractSqlStatement;
+
+public class ChangePropertyTypeDateToDatetimeStatement extends AbstractSqlStatement {
```
Comment (fuss86) [inline src/main/java/org/unbrokendome/liquibase/orientdb/statement/ChangePropertyTypeDateToDatetimeStatement.java:5]: This is not used, is it ?

**Comment by carlspring** [comment]
@evsikov ,

Thanks for your contribution!

Could you please re-indent the code from this pull request to be according to our [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html)? Thanks! :)

File: src/main/java/org/unbrokendome/liquibase/orientdb/common/OrientSqlMethod.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.unbrokendome.liquibase.orientdb.common;
+
+/**
+ * @see <a href="https://orientdb.org/docs/3.0.x/sql/SQL-Methods.html">https://orientdb.org/docs/3.0.x/sql/SQL-Methods.html</a>
+ */
+public enum  OrientSqlMethod {
+    AS_DATE, AS_DATETIME
```
Comment (fuss86) [inline src/main/java/org/unbrokendome/liquibase/orientdb/common/OrientSqlMethod.java:7]: Have you tried https://orientdb.org/docs/3.0.x/sql/SQL-Methods.html#convert ?

File: src/main/java/liquibase/change/ext/ChangePropertyTypeDateToDatetimeChange.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import javax.validation.constraints.Size;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+@Serializable(name = "changePropertyTypeDateToDatetime", description = "Change property type from date to datetime without losing data",
+        namespace = OrientSerializationConstants.ORIENT_CHANGELOG_NAMESPACE)
+public class ChangePropertyTypeDateToDatetimeChange extends AbstractMultiStatementOrientChange {
+
+    private String className;
+    private String propertyName;
```
Comment (fuss86) [inline src/main/java/liquibase/change/ext/ChangePropertyTypeDateToDatetimeChange.java:23]: this field does not match the schema. It should be `private PropertyConfig propertyConfig = new PropertyConfig();` based on your schema. See also `liquibase.change.ext.CreatePropertyChange` class.

Comment (fuss86) [reply src/main/java/liquibase/change/ext/ChangePropertyTypeDateToDatetimeChange.java:23]: Alternatively change your schema :)

**Review by fuss86 — CHANGES_REQUESTED** [review]
@evsikov once again, thank you for this PR ! This looks really good. I've left couple of comments, please consider them in your PR. This PR unleashes possibilities we have using liquibase and orientdb.

I think we should try https://orientdb.org/docs/3.0.x/sql/SQL-Methods.html#convert to make it more generic solution.

It would be great if we could do:

`UPDATE ArtifactEntry SET lastUsed = lastUsed.convert( 'datetime' ) ` making `datetime` additional property in the orient change.

Then `ChangePropertyTypeDateToDatetimeChange` could become `AlterPropertyTypeChange`

as follows 

```
class AlterPropertyTypeChange  {

    private String className;
    private String propertyName;
    private String newType;
```

(or even try to use `PropertyConfig` here for this purpose). 

Would it work ? Then we would up with a solution without `DateToDatetime` and could work for other conversions as well.

Please let us know ! Thanks.

File: src/main/java/liquibase/change/ext/ChangePropertyTypeDateToDatetimeChange.java (hunk 3 above)
Comment (evsikov) [reply src/main/java/liquibase/change/ext/ChangePropertyTypeDateToDatetimeChange.java:23]: @fuss86, thank you for all the comments here. Luckily this change is not needed anymore. I will close this PR.

**Pull request closed without merging**